### PR TITLE
Added Docker Support for development environment

### PR DESCRIPTION
**Pull request by edgar971**
Hi, 

I added a Redis a Mongo service to the docker-composer file. I also added proper configuration to the RESTFul API service in order to work with docker services. Let me know if you have any questions.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+  redis:
+    image: redis:4
+    container_name: sedaily-redis
+#    ports:
+#      - "6379:6379"
   express-mongoose-es6-rest-api:
     build:
       context: .
+    container_name: sedaily-rest-api
     volumes:
```
Comment (TheHollidayInn) [inline docker-compose.yml]: We should probably have a volume for the dev environment to link directly to the app

File: docker-compose.yml [hunk 2]
```diff
@@ -25,13 +37,17 @@ services:
     # Overwrite any env var defined in .env file (if required)
     environment:
       - MONGO_HOST=mongodb://mongo/express-mongoose-es6-rest-api-development
```
Comment (TheHollidayInn) [inline docker-compose.yml]: I'm fine with doing it this way, but we can also just add the env to the up command `--env-file` I believe.

File: gulpfile.babel.js [hunk 3]
```diff
@@ -45,7 +45,8 @@ gulp.task('nodemon', ['copy', 'babel'], () =>
     script: path.join('dist', 'index.js'),
     ext: 'js',
     ignore: ['node_modules/**/*.js', 'dist/**/*.js'],
-    tasks: ['copy', 'babel']
+    tasks: ['copy', 'babel'],
+    legacyWatch: true
```
Comment (TheHollidayInn) [inline gulpfile.babel.js]: What does this command do?

File: docker-compose.yml (hunk 1 above)
Comment (edgar971) [reply docker-compose.yml]: There is already a volume attached from host to the container. Is this what you mean? 
```yml
volumes:
      # Mounts the project directory on the host to /app inside the container,
      # allowing you to modify the code without having to rebuild the image.
      - .:/app
      # Just specify a path and let the Engine create a volume.
      # Data present in the base image at the specified mount point will be copied
      # over to the new volume upon volume initialization.
      # node_modules from this new volume will be used and not from your local dev env.
      - /app/node_modules/
```

File: gulpfile.babel.js (hunk 3 above)
Comment (edgar971) [reply gulpfile.babel.js]: In some networked environments (such as a container running nodemon reading across a mounted drive), you will need to use the legacyWatch: true which enables Chokidar's polling.

https://github.com/remy/nodemon#application-isnt-restarting

File: docker-compose.yml (hunk 2 above)
Comment (edgar971) [reply docker-compose.yml]: I moved the env variables to the .env file on my last commit.

File: docker-compose.yml (hunk 1 above)
Comment (TheHollidayInn) [reply docker-compose.yml]: Ah, my mistake, I didn't realize github had closed the section.

**Comment by TheHollidayInn** [comment]
Looks good. I'll test out later today and merge!

**Comment by TheHollidayInn** [comment]
I'm having trouble connecting to Redis:
```
events.js:160
sedaily-rest-api                 |       throw er; // Unhandled 'error' event
sedaily-rest-api                 |       ^
sedaily-rest-api                 | 
sedaily-rest-api                 | Error: Redis connection to 127.0.0.1:6379 failed - connect ECONNREFUSED 127.0.0.1:6379
sedaily-rest-api                 |     at Object.exports._errnoException (util.js:1018:11)
sedaily-rest-api                 |     at exports._exceptionWithHostPort (util.js:1041:20)
sedaily-rest-api                 |     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1090:14)
sedaily-rest-api                 | [17:36:38] [nodemon] app crashed - waiting for file changes before starting...
```

I'm going to look into this in a bit, but if you have a sec can you check?

**Comment by edgar971** [comment]
I had the same problem. Can you copy the env values from the .env.example file? There are two variables that are needed.

**Comment by TheHollidayInn** [comment]
Haha, of course. I was definitely looking at my example file. Works perfectly!

**Pull request merged**